internal/expr/parser.go

```diff
@@ -281,13 +281,27 @@ loop:
 		}
 
 		switch p.input[p.pos] {
-		// If the preceding byte is one of these then we might be at the start
-		// of an expression.
-		case ' ', '\t', '\n', '\r', '=', ',', '(', '[', '>', '<', '+', '-', '*', '/', '|', '%':
-			p.advanceByte()
+		// These bytes may be the start of an expression.
+		case '(', '*', '$', '&':
 			break loop
+		// An expression can also start with an initial name byte, e.g. a
+		// expression starting with a column name or a SQL function.
+		// Rather than testing for every initial name byte (we would stop at
+		// every letter of every word), we look for bytes that may precede the
+		// start of an expression and then check if the next byte is an initial
+		// name byte.
+		case ' ', '\t', '\n', '\r', '=', ',', '[', '>', '<', '+', '-', '/', '|', '%':
+			p.advanceByte()
+			if p.pos >= len(p.input) || isInitialNameByte(p.input[p.pos]) {
+				break loop
+			}
+			continue
 		}
+
 		p.advanceByte()
+		if p.pos == len(p.input) {
+			break loop
+		}
```

Comment:
[Nit]: This is not needed right? The condition in for loop is already checking this.